libsolidity/analysis/ContractLevelChecker.h

```diff
@@ -34,13 +36,54 @@ namespace dev
 namespace solidity
 {
 
+template <class T>
+bool hasEqualNameAndParameters(T const& _a, T const& _b)
+{
+	return _a.name() == _b.name() &&
+		FunctionType(_a).asCallableFunction(false)->hasEqualParameterTypes(
+			*FunctionType(_b).asCallableFunction(false)
+		);
+}
+
+struct LessFunction
+{
+	bool operator()(FunctionDefinition const* _a, FunctionDefinition const* _b) const
+	{
+		if (hasEqualNameAndParameters(*_a, *_b))
+			return false;
+
+		if (_a->name() != _b->name())
+			return _a->name() < _b->name();
+
```

Comment:
Everything starting from here could probably be
```
return boost::lexicographical_compare(a->parameters(), b->parameters(), [](auto const& a, auto const& b) { return a->type()->canonicalName() < b->type()->canonicalName(); });
```

Reply:
The parameters are `vector<ASTPointer<VariableDeclaration>>`, it would compare the pointers of the vector which would lead to an inconsistent ordering

Reply:
Not if you supply a compare function that *doesn't* compare pointers :-).

Reply:
ooh I didn't see you did supply a function in your example